### DISCUSSION TITLE
<CodeQL scanning (lines of code) limitation>

### DISCUSSION THREAD

**Discussion by AnilVinukonda** [community/community#48957] [Code Security] [answered]
### Select Topic Area

Question

### Body

Hello, 

We are planning to implement CodeQL for our dotnet project and would like to know if there is any limitation for scanning the lined of code.  for example : currently we are implementing SonarQube developer edition and we have a limitation of 100,000 for the SonarQube Developer Edition license. We would like to know if CodeQL has any limitations regarding code scanning with respect to lines of code. 

Thanks in advance, 
Anil Kumar.

**Comment by hmakholm** [comment, marked as answer]
There's no explicit limit on how many lines of code you can scan.

If the codebase is huge, you might run out of RAM/disk/time on the standard Actions runners, but that can be mitigated by running on beefier hardware of your own choice. (Up to a point: e.g., things begin to go bad if the total number of syntax tree nodes in the project start approaching 2^31).

**Reply by johnabroad42** [reply]
Hi how are you  pls im ask about  vip support  and make plan to travel  some  country  soon is possible  and discussion  with world  vip organization  to take action  VIP
